api/main.py

```
@@ -0,0 +1,172 @@
+def get_block(block_index: int):
+    with blockchain._lock:
+        if block_index < 0 or block_index >= len(blockchain.chain):
+            raise HTTPException(status_code=404, detail="Block not found")
+        
+        block = blockchain.chain[block_index]
+        chain_length = len(blockchain.chain)
+    
+    block_dict = block.to_dict()
+    
+    merkle_proofs = {}
+    for i, tx in enumerate(block.transactions):
```

Ruff (0.15.1)

[warning] Loop control variable `tx` not used within loop body

Rename unused `tx` to `_tx`

(B007)